ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -0,0 +1,948 @@
+                            ['M', x1, y1],
+                            ['L', x1 + width * (inverted ? -0.5 : 0.5), y1],
+                            ['L', x1 + width * (inverted ? -0.5 : 0.5), y2],
+                            ['L', x2, y2]
+                        ],
+                        this.options.link.radius
+                    )
+                };
+            }
+
+            link.dlBox = {
+                x: (x1 + x2) / 2,
+                y: (y1 + y2) / 2,
+                height: linkWidth,
+                width: 0
+            };
+            link.tooltipPos = inverted ? [
+                (this.chart.plotSizeY || 0) - link.dlBox.y,
+                (this.chart.plotSizeX || 0) - link.dlBox.x
+            ] : [
+                link.dlBox.x,
+                link.dlBox.y
+            ];
+
+        }
+    }
+    /**
+     * Private method responsible for adjusting the dataLabel options for each
+     * node-point individually.
+     */
+    public drawNodeLabels(): void {
+        let series = this,
+            mapOptionsToLevel = series.mapOptionsToLevel,
+            points = series.points,
+            options,
+            level;
+
+        points.forEach(function (point): void {
+            level = mapOptionsToLevel[point.node.level];
+            // Set options to new object to avoid problems with scope
+            options = { style: {} };
+
+            // If options for level exists, include them as well
+            if (level && level.dataLabels) {
+                options = merge(options, level.dataLabels);
+                series._hasPointLabels = true;
+            }
+
+            // Set dataLabel width to the width of the point shape.
+            if (point.shapeArgs) {
+                (options.style as any).width = point.shapeArgs.width;
+                if (point.dataLabel) {
+                    point.dataLabel.css({
+                        width: point.shapeArgs.width + 'px'
+                    });
+                }
+            }
+
+            // Merge custom options with point options
+            (point as any).dlOptions = merge(options, point.options.dataLabels);
+        });
+        Series.prototype.drawDataLabels.apply(this, arguments);
+    }
+    /**
+     * Treegraph has two separate collecions of nodes and lines,
+     * render dataLabels for both sets.
+     */
+    public drawDataLabels(): void {
+        if (this.options.dataLabels) {
+
+            if (!isArray(this.options.dataLabels)) {
+                this.options.dataLabels = [this.options.dataLabels];
+            }
+
+            const dataLabelOptions = this.options.dataLabels.map(
+                (option): DataLabelTextPathOptions | undefined =>
+                    option.textPath
+            );
+            // Render node labels.
+            this.drawNodeLabels.apply(this, arguments);
+
+            // Render link labels.
+            const points = this.points;
+            this.points = this.links as any;
```

Comment:
If this needs `any`, this calls for a fix. Links and points should have a type relation.